src/api/search_transactions.rs

```diff
@@ -97,6 +121,129 @@ impl MinaMesh {
+    // and the hash
+    let transaction_identifier = format!(
+      "{}:{}:{}:{}",
+      internal_command.command_type.to_string().to_case(Case::Snake),
+      internal_command.sequence_no,
+      internal_command.secondary_sequence_no,
+      internal_command.hash
+    );
+    let fee = internal_command.fee.unwrap_or("0".to_string());
+    let status = &internal_command.status;
+
+    let mut operations = Vec::new();
+    let mut operation_index = 0;
+
+    // Receiver Account Identifier
+    let receiver_account_id = &AccountIdentifier {
+      address: internal_command.receiver.clone(),
+      metadata: Some(json!({ "token_id": DEFAULT_TOKEN_ID })),
+      sub_account: None,
+    };
+
+    // Handle Account Creation Fee if applicable
+    if let Some(creation_fee) = &internal_command.creation_fee {
+      operations.push(operation(
+        operation_index,
+        Some(creation_fee),
+        receiver_account_id,
+        OperationType::AccountCreationFeeViaFeeReceiver,
+        Some(status),
+        None,
+        None,
+      ));
+      operation_index += 1;
+    }
+
+    match internal_command.command_type {
+      InternalCommandType::Coinbase => {
+        operations.push(operation(
+          operation_index,
+          Some(&fee),
+          receiver_account_id,
+          OperationType::CoinbaseInc,
+          Some(status),
+          None,
+          None,
+        ));
+      }
+
+      InternalCommandType::FeeTransfer => {
+        operations.push(operation(
+          operation_index,
+          Some(&fee),
+          receiver_account_id,
+          OperationType::FeeReceiverInc,
+          Some(status),
+          None,
+          None,
+        ));
+      }
+
+      InternalCommandType::FeeTransferViaCoinbase => {
+        if let Some(coinbase_receiver) = &internal_command.coinbase_receiver {
+          operations.push(operation(
+            operation_index,
+            Some(&fee),
+            receiver_account_id,
+            OperationType::FeeReceiverInc,
+            Some(status),
+            None,
+            None,
+          ));
+          operation_index += 1;
+
+          operations.push(operation(
+            operation_index,
+            Some(&fee),
+            &AccountIdentifier::new(coinbase_receiver.to_string()),
+            OperationType::FeePayerDec,
+            Some(status),
+            Some(vec![operation_index - 1]),
+            None,
+          ));
+        }
+      }
+    }
+
+    let block_identifier = BlockIdentifier::new(
+      internal_command.height.unwrap_or_default(),
+      internal_command.state_hash.unwrap_or_default(),
+    );
+    let transaction = Transaction {
+      transaction_identifier: Box::new(TransactionIdentifier::new(transaction_identifier)),
+      operations,
+      related_transactions: None,
+      metadata: None,
+    };
+
+    BlockTransaction::new(block_identifier, transaction)
+  }
+}
+
```

Comment:
This logic is common to the block endpoint. Should we extract this and reuse in the block endpoint? (the same applies for the other command types)

Reply:
That is definitely a good idea! It seems as a bit larger task though, so in order not to bloat this PR I've created a separate task to address it -> https://github.com/MinaFoundation/MinaMesh/issues/53 :+1: